src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ContentOptions.cs

```diff
@@ -29,6 +30,12 @@ public ContentPartOption AddContentPart(Type contentPartType)
             return option;
         }
 
+        public void WithContentPartFactoryType(string factoryName, Type contentPartType, Type factoryType)
+        {
+            var option = _contentParts.FirstOrDefault(x => x.Type.Name == contentPartType.Name);
+            option.WithFactoryType(factoryName, factoryType);
```

Comment:
Just did a quick look so not sure, will see more in details asap.

I think here you can just do `x.Type == contentPartType`.